### PR TITLE
Product listing filter

### PR DESCRIPTION
**Pull request by dev-ashishsethi**

### CODE REVIEW

File: productListing/productListing.css [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    -webkit-line-clamp: 3;
+    background-color: var(--color-white);
+    font-size: 15px;
+    text-align: justify;
+    word-spacing: -1px;
+    overflow: hidden;
+    margin: 1rem 0rem 1rem 1rem;
+    color: var(--color-black);
+}
+
```
Comment (m26unkwn) [inline productListing/productListing.css:97]: Everything is looking good to me.
Cross browser compatibility is good. to use.

**Review by m26unkwn — APPROVED** [review]
Cross Browser compatibility.

File: app.js [hunk 2]
```diff
@@ -1,5 +1,7 @@
 let carousel= document.querySelector("#carousel");
 
+
+
 let arrayOfImages=[
     "assets\\images\\Anime-silder-1920-1920x450.webp",
```
Comment (manojsarna) [inline app.js:6]: check spacing

File: productListing/product.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+const sliderColor= ()=>{
+    if(slider.value==="1"){
+        firstStar.style.color="#FFD700";
+        secondStar.style.color="white";
+        thirdStar.style.color="white";
+        fourthStar.style.color="white";
+        fifthStar.style.color="white";
+    }else if(slider.value==="2"){
+        firstStar.style.color="#FFD700";
+        secondStar.style.color="#FFD700";
```
Comment (manojsarna) [inline productListing/product.js:19]: good work , but try to optimize it!!

File: productListing/productListing.css [hunk 4]
```diff
@@ -0,0 +1,147 @@
+*{
+    color: var(--color-white);
+}
+.main-content{
+    display: grid;
+    grid-template-areas: "sidebar pr";
+    grid-template-columns: 0.3fr 1.5fr;
+    gap: 1rem;
+}
```
Comment (manojsarna) [inline productListing/productListing.css:9]: need to check if it is a good practice? do check!!

File: productListing/productListing.css [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+    display: block;
+    margin-bottom: 7px;
+}
+
+.label{
+    display: block;
+    color: var(--color-white);
+    margin-bottom: 7px;
+    font-size: large;
+}
```
Comment (manojsarna) [inline productListing/productListing.css:34]: looks good!!

File: productListing/productListing.html [hunk 6]
```diff
@@ -0,0 +1,265 @@
...
+<body>
+    <nav class="nav-container">
+        <a href="../index.html">
+            <p class="nav-title">R<span class="nav-title-short">&</span>B <span class="nav-title-short">Store</span>
+            </p>
+        </a>
+
+        <div class="input-box">
+            <i class="fas fa-search"></i>
+            <input type="text" class=" store-input-box" placeholder="Search the merch" id="email">
```
Comment (PuneetP16) [inline productListing/productListing.html:21]: Is it a Typo?

File: productListing/productListing.css [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+.category-filter{
+    margin: 2rem 0rem;
+}
+
+.filter{
+    display: flex;
+    justify-content: space-between;
+}
+
```
Comment (manojsarna) [inline productListing/productListing.css:47]: what this class is refering to?

**Review by PuneetP16 — APPROVED** [review]
Looks good to merge.

File: productListing/productListing.html (hunk 6 above)
Comment (dev-ashishsethi) [reply productListing/productListing.html:21]: which line are you referring to?

File: productListing/productListing.css [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+    width: 18rem;
+    background-color: var(--color-white);
+    border-radius: 1rem;
+    justify-content: center;
+    padding: 1rem 1rem 1rem 0rem;
+}
+
+.ecommerce-image{
+    width:10rem;
+    align-self: center;
```
Comment (manojsarna) [inline productListing/productListing.css:82]: looks good!!

File: productListing/productListing.css [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+.price-and-btn{
+    display: flex;
+    background-color: transparent;
+    justify-content: space-between;
+}
+
+.product-btn-section{
+    display: block;
+    margin-left:1rem ;
+    margin-top: 1rem;
```
Comment (manojsarna) [inline productListing/productListing.css:122]: looks good!!

**Review by manojsarna — APPROVED** [review]
Looks Good!! do check comments

File: productListing/product.js [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+
+const sliderColor= ()=>{
+    if(slider.value==="1"){
+        firstStar.style.color="#FFD700";
+        secondStar.style.color="white";
+        thirdStar.style.color="white";
+        fourthStar.style.color="white";
+        fifthStar.style.color="white";
+    }else if(slider.value==="2"){
+        firstStar.style.color="#FFD700";
```
Comment (muskanbagrecha) [inline productListing/product.js:18]: Declare color as a variable.

File: productListing/productListing.css [hunk 11]
```diff
@@ -0,0 +1,147 @@
...
+    margin-left: 1rem;
+    margin-top: 1rem;
+    height: 100vh;
+    grid-area: sidebar;
+}
+.section-line{
+    background-color: var(--color-white);
+    width: 100%;
+    height: 2px;
+    margin-bottom: 7px;
```
Comment (dipen2000) [inline productListing/productListing.css:21]: Use rems

File: productListing/productListing.css [hunk 12]
```diff
@@ -0,0 +1,147 @@
...
+.whishlist{
+    position: absolute;
+    top: 1rem;
+    right:1rem;
+    background-color: transparent;
+    color: var(--color-red);
+    font-size: larger;
+    cursor: pointer;
+}
+
```
Comment (dipen2000) [inline productListing/productListing.css:140]: looks good

File: productListing/productListing.css [hunk 13]
```diff
@@ -0,0 +1,147 @@
+*{
+    color: var(--color-white);
+}
+.main-content{
+    display: grid;
+    grid-template-areas: "sidebar pr";
```
Comment (muskanbagrecha) [inline productListing/productListing.css:6]: rename pr

File: productListing/productListing.html (hunk 6 above)
Comment (PuneetP16) [reply productListing/productListing.html:21]: Line 21, Search the merch.

File: productListing/productListing.css [hunk 14]
```diff
@@ -0,0 +1,147 @@
...
+.label{
+    display: block;
+    color: var(--color-white);
+    margin-bottom: 7px;
+    font-size: large;
+}
+
+.price-filter{
+    margin-top: 2rem;
+}
```
Comment (muskanbagrecha) [inline productListing/productListing.css:38]: add another line break for consistency

File: productListing/productListing.html [hunk 15]
```diff
@@ -0,0 +1,265 @@
...
+            <div class="product-card">
+                <img src="..\assets\images\products\Marvel\image_40-removebg-preview.webp" alt="" class="ecommerce-image">
+                    
+                    <i class="far fa-heart whishlist"></i>
+                    <h4 class="product-title">FUNKO POP!</h4>
+                    <p class="ecommerce-description">POP TV: Squid Game- Masked Worker Pre-Order Lorem ipsum dolor sit
+                        amet
+                        consectetur, adipisicing elit. Animi, amet.</p>
+                        <div class="price-and-btn">
+                            <strong class="ecommerce-price">₹ 999</strong>
```
Comment (dipen2000) [inline productListing/productListing.html:166]: I like that you have used strong tag here to make it semantic, I used p tag for that and that is not semantically correct for this usecase. Looks good.

File: productListing/productListing.css [hunk 16]
```diff
@@ -0,0 +1,147 @@
...
+.rating-container{
+    display: flex;
+    flex-direction: column;
+}
+.slider{
+    width: 10rem;
+}
+.star{
+    margin-right: 0.7rem;
+
```
Comment (muskanbagrecha) [inline productListing/productListing.css:62]: remove extra space

File: signIn and SignUp/signUp.html [hunk 17]
```diff
@@ -66,6 +66,22 @@ <h3 class="form-title">SignUp</h3>
...
+            </li>
+            <li class="social-links">
+                <a href="https://twitter.com/web_AshishSethi" target="_blank"><svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" role="img" class="iconify iconify--mdi" width="32" height="32" preserveAspectRatio="xMidYMid meet" viewBox="0 0 24 24"><path d="M22.46 6c-.77.35-1.6.58-2.46.69c.88-.53 1.56-1.37 1.88-2.38c-.83.5-1.75.85-2.72 1.05C18.37 4.5 17.26 4 16 4c-2.35 0-4.27 1.92-4.27 4.29c0 .34.04.67.11.98C8.28 9.09 5.11 7.38 3 4.79c-.37.63-.58 1.37-.58 2.15c0 1.49.75 2.81 1.91 3.56c-.71 0-1.37-.2-1.95-.5v.03c0 2.08 1.48 3.82 3.44 4.21a4.22 4.22 0 0 1-1.93.07a4.28 4.28 0 0 0 4 2.98a8.521 8.521 0 0 1-5.33 1.84c-.34 0-.68-.02-1.02-.06C3.44 20.29 5.7 21 8.12 21C16 21 20.33 14.46 20.33 8.79c0-.19 0-.37-.01-.56c.84-.6 1.56-1.36 2.14-2.23z" fill="currentColor"></path></svg></a>
+            </li>
+        </ul>
+    </footer>
```
Comment (dipen2000) [inline signIn and SignUp/signUp.html:85]: looks good.

**Review by dipen2000 — APPROVED** [review]
Looks good to me.

File: productListing/productListing.css [hunk 18]
```diff
@@ -0,0 +1,147 @@
...
+.product-card{
+    position: relative;
+    display: flex;
+    flex-direction: column;
+    width: 18rem;
+    background-color: var(--color-white);
+    border-radius: 1rem;
+    justify-content: center;
+    padding: 1rem 1rem 1rem 0rem;
```
Comment (muskanbagrecha) [inline productListing/productListing.css:77]: Use shorthand properties top right-left bottom

File: productListing/productListing.css [hunk 19]
```diff
@@ -0,0 +1,147 @@
...
+    margin-top: 1rem;
+    /* width: 16rem; */
+    border-radius: 0.3rem;
+}
+
+.w-100{
+    width: 100%;
+    
+}
+.whishlist{
```
Comment (muskanbagrecha) [inline productListing/productListing.css:131]: fix spelling

File: productListing/productListing.css [hunk 20]
```diff
@@ -0,0 +1,147 @@
...
+    font-size: larger;
+    cursor: pointer;
+}
+
+.product-listing{
+    display: flex;
+    flex-wrap: wrap;
+    gap: 4rem;
+    height: fit-content;
+    margin: 4rem 0rem 0rem 6rem;
```
Comment (muskanbagrecha) [inline productListing/productListing.css:146]: Use shorthand property

File: style.css [hunk 21]
```diff
@@ -192,4 +195,5 @@ a:visited{
 .social-links{
     float: left;
```
Comment (muskanbagrecha) [inline style.css:196]: Can use some other alternative to push it to the left as float creates a stacking context

**Review by muskanbagrecha — CHANGES_REQUESTED** [review]
Increase spacing below clear button.

**Pull request merged**